File: strategies/suicidal_python/strategies.py
#!/usr/bin/env python

import sys

isAnt = False

for line in sys.stdin:
    line = line.strip()
    if line == "END":
        if isAnt:
            print("SUICIDE")
        print("END", flush=True)
        isAnt = False
    elif line == "BEGIN ANT":
        isAnt = True

File: strategies/suicidal_perl/strategies.pl
#!/usr/bin/env perl

my $isAnt = 0;
while ($var = <STDIN>) {
	if ($var eq "END\n") {
		if ($isAnt) {
			print "SUICIDE\n";
		}
		print "END\n";
		$| = 1;
		$isAnt = 0;
	} elsif ($var eq "BEGIN ANT\n") {
		$isAnt = 1;
	}
}
